src/blocks/focused_window.rs

```diff
@@ -0,0 +1,180 @@
+use serde::de::Deserialize;
+use serde_derive::Deserialize;
+use swayipc_async::{Connection, Event, EventType, Node, WindowChange, WorkspaceChange};
+use tokio::sync::mpsc;
+use tokio_stream::StreamExt;
+
+use crate::blocks::{BlockEvent, BlockMessage};
+use crate::config::SharedConfig;
+use crate::errors::{BlockError, Result, ResultExt};
+use crate::widgets::text::TextWidget;
+use crate::widgets::I3BarWidget;
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum MarksType {
+    All,
+    Visible,
+    None,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
+pub struct FocusedWindowConfig {
+    /// Truncates titles if longer than max-width
+    pub max_width: usize,
```

Comment:
Can you please add `format` config option? It would make it possible to set both max width as well as min width

Reply:
Or just a TODO :smile: